File: conformance/code/employee-go/internal/models/employee_patch.go
package models

// EmployeePatch carries an employee update — the same shape serves the PATCH
// request body and the persistence-layer change set. Nil pointer fields are
// skipped by GORM's Updates(struct); a pointer to a zero value (e.g. *false,
// *"") is written as-is.
//
// Only fields on this struct are updatable. Immutable attributes — id, code,
// userId, individualId, dateOfAppointment, tenantId, jurisdictions, and
// audit columns — are intentionally absent and cannot be rewritten through
// the update path.
//
// Audit fields are hidden from JSON so clients cannot spoof them; they are
// set by the service on every update.
type EmployeePatch struct {
	Status       *string `json:"status,omitempty"`
	EmployeeType *string `json:"employeeType,omitempty"`
	Department   *string `json:"department,omitempty"`
	Designation  *string `json:"designation,omitempty"`
	IsActive     *bool   `json:"isActive,omitempty"`

	// Version is the bumped optimistic-concurrency value (expected+1) written on
	// every patch. It is a plain int (not a pointer) so GORM's Updates(struct)
	// always writes it — the CAS predicate (WHERE version = expected) is applied
	// by the repository. json:"-" so clients can't spoof it via the patch body;
	// the client's expected version travels on PatchEmployeeRequest.Version.
	Version int `json:"-" gorm:"column:version"`

	// See Employee entity — plain form, GORM auto-quotes case-sensitive
	// identifiers. The escape form (column:\"modifiedBy\") only matters for
	// SELECT mapping, but kept consistent here for clarity.
	ModifiedBy   string `json:"-" gorm:"column:modifiedBy"`
	ModifiedTime int64  `json:"-" gorm:"column:modifiedTime"`
}

File: conformance/code/employee/src/main/java/com/digit/employee/model/EmployeePatch.java
package com.digit.employee.model;

/**
 * Persistence change-set for PATCH (mirrors Go {@code EmployeePatch}). Null fields are skipped by the
 * repository's partial update; a non-null value (including {@code false}/empty) is written as-is.
 * Immutable attributes and jurisdictions are intentionally absent. Audit fields are server-set.
 */
public class EmployeePatch {

    private String status;
    private String employeeType;
    private String department;
    private String designation;
    private Boolean isActive;
    // Bumped optimistic-concurrency value (expected+1), written on every patch. The CAS predicate
    // (WHERE version = expected) is applied by the repository.
    private int version;
    private String modifiedBy;
    private long modifiedTime;

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getEmployeeType() { return employeeType; }
    public void setEmployeeType(String employeeType) { this.employeeType = employeeType; }
    public String getDepartment() { return department; }
    public void setDepartment(String department) { this.department = department; }
    public String getDesignation() { return designation; }
    public void setDesignation(String designation) { this.designation = designation; }
    public Boolean getIsActive() { return isActive; }
    public void setIsActive(Boolean isActive) { this.isActive = isActive; }
    public int getVersion() { return version; }
    public void setVersion(int version) { this.version = version; }
    public String getModifiedBy() { return modifiedBy; }
    public void setModifiedBy(String modifiedBy) { this.modifiedBy = modifiedBy; }
    public long getModifiedTime() { return modifiedTime; }
    public void setModifiedTime(long modifiedTime) { this.modifiedTime = modifiedTime; }
}
